.buildkite/scripts/generate_pipeline.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "--- Install dependencies"
+pip3 install --quiet jinja2
+pip3 install --quiet PyYAML
+
+echo "--- Run pipeline generator in dry-run mode"
+python3 .buildkite/pipeline.py || true
```

Comment:
I suggest we also print the yaml output with something like `python3 .buildkite/pipeline.py | yq .` at least during development to help troubleshoot. `yq` is available in our prebuild images, otherwise it can be easily installed on the container where this step runs via

```
curl -fsSL --retry-max-time 60 --retry 3 --retry-delay 5 -o /usr/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
chmod a+x /usr/bin/yq
```